lib/engine/game/base.rb

```diff
@@ -2891,6 +2895,10 @@ def next_sr_position(entity)
                            @players.sort_by { |p| [p.cash, @players.index(p)] }.reverse
                          when :least_cash
                            @players.sort_by { |p| [p.cash, @players.index(p)] }
+                         when :next_clockwise
+                           @round.next_entity_index
+                         when :most_cash_keep_order
+                           @players.max_by(&:cash)
```

Comment:
This method returns an array of player objects. You're returning either a number or a single player object. This should probably be something like:
```suggestion
                         when :next_clockwise
                           @players.rotate
                         when :most_cash_keep_order
                           @players.rotate(@players.index(@players.max_by(&:cash)))
```
However, I don't think that this code will get called unless you add `:next_clockwise` and `:max_cash_keep_order` here:
https://github.com/tobymao/18xx/blob/f6a04a92260bb3c0af5aeb2022457f1d417d0166/assets/app/view/game/player.rb#L193-L194

Reply:
Hey Olly, I have a couple of questions. First, I'm not sure what you mean with regards to the /assets/app/view/game/player.rb part. 

For the first part above, your suggested code works on it's own. Mine did too, but I'm curious why yours is an improvement.

Reply:
The only place that `Game.next_sr_position` is called is in `View::Game::Player`, and the call to `next_sr_position` is guarded by a check that `Game.next_sr_player_order` is one of `:first_to_pass`, `:most_cash` or `:least_cash`.

Here, `next_sr_player_order` will return either `:most_cash_keep_order` or `:next_clockwise`, so this version of `next_sr_position` is never being called, which is why you're not seeing any errors.

Reply:
Gotcha! Done.